canopen/objectdictionary/__init__.py

```diff
@@ -387,7 +395,7 @@ def decode_raw(self, data: bytes) -> Union[int, float, str, bytes, bytearray]:
             return data.rstrip(b"\x00").decode("ascii", errors="ignore")
         elif self.data_type == UNICODE_STRING:
             # Is this correct?
-            return data.rstrip(b"\x00").decode("utf_16_le", errors="ignore")
+            return data.decode("utf_16_le", errors="ignore").rstrip("\x00")
```

Comment:
What's this change about?  Please adjust the comment accordingly.

It does make sense IMHO to strip a trailing NUL character, and after decoding that's more sensible.  The type is defined as `ARRAY [length] OF UNSIGNED16`, thus a single NUL cannot be trailing.  And if the last byte happens to be a NUL, then removing it definitely breaks decoding the last glyph.  If there is geniunely a single trailing NUL byte in the input, it will be ignored anyway by the decode option.

Reply:
BTW: Technically NUL is not specified as termination of string in the canopen standard, so this is a feature which the canopen python module adds on top. This is the same choice that it use utf_16_le, which is also not specified in the standard.

Reply:
Yes, I'm aware of that.  Still it will be common to get NUL-terminated strings from C-based device implementations, and we don't want that terminator to leak into Python strings.  Thus it makes sense to do it like this.